jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/HttpConnectionTest.java

```diff
@@ -1724,4 +1727,98 @@ public boolean handle(Request request, Response response, Callback callback) thr
         else
             assertThat(response.get(HttpHeader.CONNECTION), is(expectedConnectionHeader));
     }
+
+    @Test
+    public void testRetainedChunks() throws Exception
+    {
+        Queue<Content.Chunk> chunks = new ConcurrentLinkedQueue<>();
+        CountDownLatch blocked = new CountDownLatch(1);
+
+        _server.setHandler(new Handler.Abstract()
+        {
+            @Override
+            public boolean handle(Request request, Response response, Callback callback)
+            {
+                while (true)
+                {
+                    Content.Chunk chunk = request.read();
+                    if (chunk == null)
+                    {
+                        try
+                        {
+                            blocked.countDown();
+                            CountDownLatch blocker = new CountDownLatch(1);
+                            request.demand(blocker::countDown);
```

Comment:
Use `Blocker.runnable()` instead!

This is a habit we must get in our muscle memory as the `Runnable` passed to `demand()` must be non-blocking!